fluminum/app/(tabs)/spirit-level.tsx

```diff
@@ -1,22 +1,85 @@
 import * as React from "react";
 import { View } from "react-native";
-import { Text } from "~/components/ui/text";
+import {useEffect, useMemo, useRef, useState} from "react";
+import {DeviceMotion} from "expo-sensors";
+import Svg, {Text,Line, Polygon} from "react-native-svg";
+import {Text as UiText} from "~/components/ui/text"
+import {Button} from "~/components/ui/button";
 
 export default function Screen() {
+    const [{beta, gamma}, setData] = useState({
+        beta: 0,
+        gamma: 0,
+    });
+    const [{betaDisplay,gammaDisplay}, setDisplayData] = useState({
+        betaDisplay: 0,
+        gammaDisplay: 0,
+    })
+    const betaOfset = useRef(0)
+    const gammaOfset = useRef(0)
+
+    const [pointsTop, setPointsTop] = useState("");
+    const [pointsLeft, setPointsLeft] = useState("");
+    useEffect(() => {
+        DeviceMotion.setUpdateInterval(25)
+        console.log("hi")
+        const listener = DeviceMotion.addListener(data => {
+            if(data && data.rotation) {
+                const localData = {beta: 0, gamma: 0};
+                localData.beta = data.rotation.beta - betaOfset.current
+                localData.gamma = data.rotation.gamma - gammaOfset.current
+                setData(localData)
+                const localDisplayData = {betaDisplay: 0, gammaDisplay: 0,}
+                localDisplayData.betaDisplay = (180 / Math.PI * (data.rotation.beta-betaOfset.current))
+                localDisplayData.gammaDisplay = (180 / Math.PI * (data.rotation.gamma-gammaOfset.current))
+                setDisplayData(localDisplayData)
```

Comment:
something like this would be cleaner imo
```ts
const displayData = useMemo(
  () => ({
    beta: (180 / Math.PI) * (data.rotation.beta - betaOffset.current),
    gamma: (180 / Math.PI) * (data.rotation.gamma - gammaOffset.current),
  }),
  [data],
);
```